nixos/modules/services/misc/omnom.nix

```diff
@@ -0,0 +1,281 @@
+                default = true;
+                description = "Whether to limit cookies to a secure channel.";
+              };
+            };
+            storage = {
+              type = lib.mkOption {
+                type = lib.types.str;
+                default = "fs";
+                description = "Storage type.";
+              };
+              root = lib.mkOption {
+                type = lib.types.path;
+                default = "${cfg.dataDir}/static/data";
+                defaultText = lib.literalExpression ''
+                  "''${config.services.omnom.dataDir}/static/data"
+                '';
+                description = "Where the snapshots are saved.";
+              };
+            };
+            smtp = {
+              tls = lib.mkEnableOption "Whether TLS encryption should be used.";
+              tls_allow_insecure = lib.mkEnableOption "Whether to allow insecure TLS.";
+              host = lib.mkOption {
+                type = lib.types.str;
+                default = "";
+                description = "SMTP server hostname.";
+              };
+              port = lib.mkOption {
+                type = lib.types.port;
+                default = 25;
+                description = "SMTP server port address.";
+              };
+              sender = lib.mkOption {
+                type = lib.types.str;
+                default = "Omnom <omnom@127.0.0.1>";
+                description = "Omnom sender e-mail.";
+              };
+              send_timeout = lib.mkOption {
+                type = lib.types.int;
+                default = 10;
+                description = "Send timeout duration in seconds.";
+              };
+              connection_timeout = lib.mkOption {
+                type = lib.types.int;
+                default = 5;
+                description = "Connection timeout duration in seconds.";
+              };
+            };
+          };
+        };
+        default = { };
+      };
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = !lib.hasAttr "username" cfg.settings.smtp;
+        message = ''
+          `services.omnom.settings.smtp.username` must be defined in `services.omnom.secretsFile`.
+        '';
+      }
+      {
+        assertion = !lib.hasAttr "password" cfg.settings.smtp;
+        message = ''
+          `services.omnom.settings.smtp.password` must be defined in `services.omnom.secretsFile`.
+        '';
+      }
+      {
+        assertion = !(cfg.settings.storage.root != "${cfg.dataDir}/static/data");
+        message = ''
+          For Omnom to access the snapshots, it needs the storage root
+          directory to be inside the service's working directory.
+
+          As such, `services.omnom.settings.storage.root` must be the same as
+          `''${services.omnom.dataDir}/static/data`.
+        '';
+      }
+    ];
+
+    environment.etc."omnom/config.yml".source = configFile;
```

Comment:
Let's remove this as we are keeping the config in the state directory, this is just going to confuse folks.

Reply:
Users won't be able to read the config in the state dir, but sure.